app/src/main/java/com/aboolean/movies/data/repository/PopularMoviesRepository.kt

```diff
@@ -36,6 +42,25 @@ class PopularMoviesRepositoryImpl(private val movieEndpoints: MovieEndpoints,
                 }
     }
 
+    override suspend fun suspendGetPopular(page: Int): List<Movie> = withContext(dispatcherProvider.io()){
+        val localMovies = moviesDao.suspendGetPopularLocalMovies(page)
+        val remoteMovies = try {
+            movieEndpoints.suspendGetPopularMovies(page)
+        } catch (ex : Exception) {
+            null
+        }
+
+        if (remoteMovies != null){
+            suspendSaveOnDataBase(remoteMovies)
+        }
+
+        if (localMovies.isEmpty() && remoteMovies != null){
+            sortMovies(remoteMovies.results).orEmpty()
+        } else {
+            sortMovies(localMovies).orEmpty()
+        }
```

Comment:
```suggestion
      runCatching {
            movieEndpoints.suspendGetPopularMovies(page)
        }.getOrNull()?.let { remoteMovies->
            suspendSaveOnDataBase(remoteMovies)
            if(localMovies.isEmpty()) {
                sortMovies(remoteMovies.results)
            }
            else {
                sortMovies(localMovies)
            }
        }.orEmpty().also { 
            sortMovies(localMovies)
        }
```
I would suggest this approach, what do you think? In my opinion it looks cleaner

Reply:
Yes of course, its clear and legible. I am going to commit your suggestion.